docs/en_us/ingress/rewrite.md

```diff
@@ -0,0 +1,247 @@
+| callbackList[]        | A list of rewrite rule, each rule must have an unique callback point. |
+| callbackList[].params | Params of rewrite rule.                                      |
+| callbackList[].order  | The order of rewrite rule,  effective on the same callback point. |
+| callbackList[].when   | The callback point of rewrite url, default is `AfterLocation` (only support this callback point) |
+
+## Rewrite Host
+
+The BFE Ingress Controller supports for configuring host information by annotations, including set static host or dynamic host.
+
+### Static Host
+
+Use `bfe.ingress.kubernetes.io/rewrite-url.host` to configure the rewrite action of setting host to specified value.
+
+For example：
+
+```yaml
+bfe.ingress.kubernetes.io/rewrite-url.host: '[{"params": "baidu.com", "when": "AfterLocation"}]'
+```
+
+Corresponding scenario:
+
+* Original url: http://host/path?query-key=value
+* Modified url: http://www.baidu.com/path?query-key=value
+
+### Dynamic Host
+
+Use `bfe.ingress.kubernetes.io/rewrite-url.host-from-path-prefix` to set host to specified path prefix.
+
+For example：
+
+```yaml
+bfe.ingress.kubernetes.io/rewrite-url.host-from-path-prefix: '[{"params": "true"}]'
+```
+
+Corresponding scenario:
+
+- Original url: https://old-host/new-host/path?query-key=value
+- Modified url: https://new-host/path?query-key=value
+
+## Rewrite Path
+
+The following are allowed ways to configure the rewrite actions of url path:  
+
+* Static path
+
+* Dynamic path, including add, delete and strip path's prefix.
+
+### Static Path
+
+Use `bfe.ingress.kubernetes.io/rewrite-url.path`  to set path to specified value.
+
+For example:
+
+```yaml
+bfe.ingress.kubernetes.io/rewrite-url.path: '[{"params": "/index"}]'
+```
+
+Corresponding scenario:
+
+- Original url: http://host/path?query-key=value
+- Modified url: http://host/index?query-key=value
+
+### Dynamic Path
+
+Modify the prefix of url path, including add, delete and strip prefix. Be careful about setting the `order` value in the rewrite rule.
+
+#### Add Path Prefix 
+
+Use `bfe.ingress.kubernetes.io/rewrite-url.path-prefix-add` to add prefix to original path.
+
+For example：
+
+```yaml
+bfe.ingress.kubernetes.io/rewrite-url.path-prefix-add: '[{"params": "/foo/"}]'
+```
+
+Corresponding scenario:
+
+- Original url: https://host/path?query-key=value
+- Modified url:  https://host/foo/path?query-key=value
+
+#### Delete Path Prefix
+
+Use `bfe.ingress.kubernetes.io/rewrite-url.path-prefix-trim` to trim prefix from original path.
+
+For example：
+
+```yaml
+bfe.ingress.kubernetes.io/rewrite-url.path-prefix-trim: '[{"params": "/foo/"}]'
+```
+
+Corresponding scenario:
+
+* Original url: https://host/foo/path?query-key=value
+
+- Modified url: https://host/path?query-key=value
+
+#### Strip Path Prefix
+
+Use `bfe.ingress.kubernetes.io/rewrite-url.path-prefix-strip` to strip the segments of original path.
```

Comment:
Use `bfe.ingress.kubernetes.io/rewrite-url.path-prefix-strip` to strip the indicated number of prefix segments from original path.